### PR TITLE
fix(db): use schema version as tag only for `trivy-db` and `trivy-java-db` registries by default

### LINKED ISSUE #6218: Trivy image with  flag --java-db-repository is adding ":1" to the URL of container repo
### Discussed in https://github.com/aquasecurity/trivy/discussions/6202

<div type='discussions-op-text'>

<sup>Originally posted by **KMikkey** February 26, 2024</sup>
### Description

After use of below command
`trivy image -d  --java-db-repository gitlab.example.com:4444/app/core-app/trivy-java-db:latest`
this error appear
`ERROR	Unable to initialize the Java DB: Java DB update failed: Java DB update error: DB download error: repository name error (gitlab.example.com:4444/app/core-app/trivy-java-db:latest:1): could not parse reference: gitlab.example.com:4444/app/core-app/trivy-java-db:latest:1`

### Desired Behavior

Download of java-db from repository.

### Actual Behavior

Download of java-db is not possible due to add of ":1" to the URL.

### Reproduction Steps

```bash
1.
2.
3.
...
```

### Target

Container Image

### Scanner

Vulnerability

### Output Format

JSON

### Mode

Standalone

### Debug Output

```bash
2024-02-22T14:49:49.707Z	INFO	JAR files found
2024-02-22T14:49:49.707Z	INFO	Java DB Repository: gitlab.example.com:4444/app/core-app/trivy-java-db:latest:1
2024-02-22T14:49:49.707Z	INFO	Downloading the Java DB...
2024-02-22T14:49:49.707Z	ERROR	Unable to initialize the Java DB: Java DB update failed: Java DB update error: DB download error: repository name error (gitlab.example.com:4444/app/core-app/trivy-java-db:latest:1): could not parse reference: gitlab.example.com:4444/app/core-app/trivy-java-db:latest:1
2024-02-22T14:49:49.738Z	FATAL	image scan error:
```

### Operating System

CentOS7

### Version

```bash
Version: 0.40.0
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)</div>

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
We add schema version as tag only for all registries (both default and external).
To avoid limiting users, we only need to do this for the default registries (`ghcr.io/aquasecurity/trivy-db` and `ghcr.io/aquasecurity/trivy-java-db`).

## Related issues
- Close #6218

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Review by knqyf263 — COMMENTED** [review]
If I understand correctly, the current change will be breaking compatibility. Those who specify `--db-repository myregistry` will see an error after this PR since their repository is now `myregistry:2`, but Trivy accesses `myregistry`. Am I correct?
What if adding a tag when it's not specified?

- `--db-repository ghcr.io/aquasecurity/trivy-db` => Access `ghcr.io/aquasecurity/trivy-db:2`
- `--db-repository ghcr.io/aquasecurity/trivy-db:2` => Access `ghcr.io/aquasecurity/trivy-db:2`
- `--db-repository myregistry` => Access `myregistry:2`
- `--db-repository myregistry:2` => Access `myregistry:2`
- `--db-repository myregistry:3` => Access `myregistry:3`

**Comment by DmitriyLewen** [comment]
@knqyf263 I updated this PR using your solution
Take a look, when you have time, please.

**Comment by knqyf263** [comment]
And the default value should be `ghcr.io/aquasecurity/trivy-db:2` now.

**Comment by DmitriyLewen** [comment]
You are right.
Changed in https://github.com/aquasecurity/trivy/pull/6219/commits/f4048b7b779f55cd7c337f2159fb60521a7313f4

File: pkg/db/db.go [hunk 1]
```diff
@@ -78,6 +79,11 @@ func NewClient(cacheDir string, quiet bool, opts ...Option) *Client {
 		opt(o)
 	}
 
+	// Add the schema version as a tag if the tag doesn't exist.
```
Comment (knqyf263) [inline pkg/db/db.go:82]: Can you add a comment this is for backward compatibility? We want to remove it after a while.

File: pkg/javadb/client.go [hunk 2]
```diff
@@ -82,8 +83,12 @@ func (u *Updater) Update() error {
 }
 
 func Init(cacheDir, javaDBRepository string, skip, quiet bool, registryOption ftypes.RegistryOptions) {
+	// Add the schema version as a tag if the tag doesn't exist.
```
Comment (knqyf263) [inline pkg/javadb/client.go:86]: ditto

**Review by knqyf263 — APPROVED** [review]
I left small comments, but LGTM basically.

File: pkg/db/db.go (hunk 1 above)
Comment (DmitriyLewen) [reply pkg/db/db.go:82]: Added comment in https://github.com/aquasecurity/trivy/pull/6219/commits/91bcb0ab8ce79d6e58ec07600ee10fd2287fe4f7

File: pkg/javadb/client.go (hunk 2 above)
Comment (DmitriyLewen) [reply pkg/javadb/client.go:86]: Added comment in https://github.com/aquasecurity/trivy/pull/6219/commits/91bcb0ab8ce79d6e58ec07600ee10fd2287fe4f7

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**